controller/hybridgateway/controller.go

```diff
@@ -35,13 +35,16 @@ import (
 // RootObjectPtr interfaces, allowing flexible reconciliation logic for different resource types.
 type HybridGatewayReconciler[t converter.RootObject, tPtr converter.RootObjectPtr[t]] struct {
 	client.Client
+	// ReferenceGrantEnabled indicates whether ReferenceGrants are enabled in the cluster.
```

Comment:
nit: I would be verbose on the comment about `ReferenceGrant` be enabled/disabled automatically depending on their CRD availability in the cluster.
```suggestion
	// ReferenceGrantEnabled indicates whether ReferenceGrants are enabled in the cluster (i.e., the CRD is available).
```